include/zephyr/net/lwm2m.h

```diff
@@ -1378,5 +1397,18 @@ int lwm2m_engine_send(struct lwm2m_ctx *ctx, char const *path_list[], uint8_t pa
  */
 struct lwm2m_ctx *lwm2m_rd_client_ctx(void);
 
+/** 
+ * @brief LwM2M Resource data cache enable
+ *
+ * @param resource_path LwM2M resourcepath string "obj/obj-inst/res(/res-inst)"
+ * @param data_cache Pointer to Data cache array
+ * @param cache_element_size Length of Data cache entries
+ * 
+ * @return 0 for success or negative in case of error.
+ *
+ */
+int lwm2m_engine_enable_cache(char const *resource_path, struct lwm2m_time_series_elem *data_cache,
+			      size_t cache_element_size);
```

Comment:
```suggestion
			      size_t cache_len);
```

Maybe the length of cache should be named `len`. Then document should specify if it is bytes or `lwm2m_time_series_elem` elements.